include/meta/meta.hpp

```diff
@@ -2837,6 +2841,63 @@ namespace meta
     /// \endcond
 #endif
 
+#if META_CXX_VISIT
+    ///////////////////////////////////////////////////////////////////////////////////////////
+    // visit
+    /// \cond
+    namespace detail
+    {
+        
+        struct visit_fn
+        {
+            template<typename Fn, typename Type>
+            struct visit{
+                static constexpr void call(Fn f){ f(Type{}); }
+            };
+
+            template<typename Fn>
+            using FnPtr = decltype(&visit<Fn, void>::call);
+
+            template<typename Fn, typename... Types>
+            static constexpr auto make_fn_map(list<Types...>) -> std::array<FnPtr<Fn>, sizeof...(Types)>
+            {
+                return {&visit<Fn, Types>::call...};
+            }
+            
+            template <class Fn, class... Args>
+            constexpr auto operator()(list<Args...> l, std::size_t i, Fn&& f) const -> Fn
+            {
+                using F = decltype(std::forward<Fn>(f));
+                constexpr auto map = make_fn_map<F>(l);
+                if(i < map.size()) map[i](std::forward<Fn>(f));
+                return f;
+            }
+        };
+    }  // namespace detail
+    /// \endcond
+
+    #if META_CXX_INLINE_VARIABLES
+        /// `visit(List, I, Fn)` calls the \p UnaryFunction for the \p i 
+        /// -th argument in the \p List. No-opt if \p i not in range.
```

Comment:
I did not want to throw exception here -> no-op if not in range. Should we make that UB and leave the check to the user?